frontend/src/editor/transcription_editor.tsx

```diff
@@ -63,14 +64,29 @@ function renderElement({ element, children, attributes }: RenderElementProps): J
 }
 
 function renderLeaf({ leaf, children, attributes }: RenderLeafProps): JSX.Element {
+  const parent: Paragraph = children.props.parent;
+  const myIndex = parent.children.findIndex((x) => x.start == leaf.start);
+  let wordStartIndex = myIndex;
+  for (
+    ;
+    wordStartIndex > 0 && !parent.children[wordStartIndex].text.startsWith(' ');
+    wordStartIndex--
+  );
+  const wordStart = parent.children[wordStartIndex];
+
   const classes = ['word'];
-  if (leaf.conf != undefined && leaf.conf < 0.7) {
-    classes.push('text-red-600 dark:text-red-500');
-  }
-  if (leaf.start !== undefined) {
-    classes.push(startTimeToClassName(leaf.start));
+  if (wordStart?.start !== undefined) {
+    classes.push(startTimeToClassName(wordStart.start));
   }
 
+  const systemPrefersDark = useMediaQuery('(prefers-color-scheme: dark)');
+
+  const color = !leaf.conf
+    ? undefined
+    : systemPrefersDark
+    ? `hsl(0, 100%, ${Math.min(leaf.conf * 50 + 50, 100)}%)`
+    : `hsl(0, 100%, ${Math.min((1 - leaf.conf) * 100, 50)}%)`;
```

Comment:
```suggestion
    : `hsl(0, 100%, ${Math.min((1 - leaf.conf) * 100, 45)}%)`;
```

if we do this change we get the required contrast ratios for both light and dark